Makefile

```diff
@@ -103,6 +106,30 @@ deploy: manifests kustomize ## Deploy controller to the K8s cluster specified in
 undeploy: ## Undeploy controller from the K8s cluster specified in ~/.kube/config.
 	$(KUSTOMIZE) build config/default | kubectl delete -f -
 
+olm-manifests: manifests ## Copy manifests from config/ to manifests/
+	cp -f config/crd/bases/externaldns.olm.openshift.io_externaldnses.yaml manifests/externaldns.olm.openshift.io_crd.yaml
+	cp -f config/rbac/role.yaml manifests/external-dns-operator_rbac.authorization.k8s.io_v1_clusterrole.yaml
+	cp -f config/rbac/role_binding.yaml manifests/external-dns-operator_rbac.authorization.k8s.io_v1_clusterrolebinding.yaml
+	cp -f config/rbac/service_account.yaml manifests/external-dns-operator_v1_serviceaccount.yaml
+	cp -f config/rbac/auth_proxy_role.yaml manifests/external-dns-operator-auth-proxy_rbac.authorization.k8s.io_v1_clusterrole.yaml
+	cp -f config/rbac/auth_proxy_role_binding.yaml manifests/external-dns-operator-auth-proxy_rbac.authorization.k8s.io_v1_clusterrolebinding.yaml
+	cp -f config/rbac/auth_proxy_service.yaml manifests/external-dns-operator-auth-proxy_v1_service.yaml
```

Comment:
I'm a little confused how all the pieces fit together, why `manifests/` has a file that isn't copied from `config/`, and why `config/` has files that don't appear to be incorporated into the OLM bundle anywhere.  Do we need anything from `config/manager/`?  Should `manifests/external-dns-operator_clusterserviceversion.yaml` be copied in from somewhere?  Should `config/rbac/externaldns_editor_role.yaml` and `config/rbac/externaldns_viewer_role.yaml` be copied in as well?

Reply:
Normally `operator-sdk` is supposed do all the job:
- Create `ClusterServiceVersion` manifest with `operator-sdk generate kustomize manifests` command
- Get the output of your manifests from `config/default/` and generate a bundle (manifests + ClusterServiceVersion + metadata + Dockerfile) out of it with `operator-sdk generate bundle` command

But I decided to not add `operator-sdk` to the repository for these reasons:
- it's a kubebuilder project, so to not add confusion about why operator-sdk is used in one place but not in others
- operator-sdk generation of ClusterServiceVersion is not optimal: it needs some parameters from you but not all you want is parameterized. Some things are defaulted and don't work in your case some has be added/changed in order for `opm` to pass the checks.
- operator-sdk generation of some other things is not optimal for this project, e.g. unused scorecard stuff is added.

So, overall I found `operator-sdk generate bundle` is good for the scaffolding but not for the regular regeneration. That's why I put this small make rule to simplify a little the life of creation of OLM bundle. As you rightfully noticed CSV is not touched by this command. I let it to be modified manually as it's complex therefore a bit difficult to be automated.

The contents of `config/manager` is in CSV for the install strategy part. I moved all the other resources to the dedicated files except for the deployment as the install strategy has to contain something.

Will update the comment with more details.